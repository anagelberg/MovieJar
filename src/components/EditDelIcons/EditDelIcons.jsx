import './EditDelIcons.scss';
import delIcon from "../../assets/icons/delete.svg";
import editIcon from "../../assets/icons/edit.svg";


function EditDelIcons({ delClick, editClick }) {
    return (
        <div className="icons">
            <img className="icons__icon" src={editIcon} alt="pencil icon" />
            <img className="icons__icon" src={delIcon} alt="trashcan icon" onClick={delClick} />
        </div>
    )
}

export default EditDelIcons;