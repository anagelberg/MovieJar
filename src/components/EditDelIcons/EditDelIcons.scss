@use "../../styles/_mixins.scss" as *;
@use "../../styles/_variables.scss" as *;

.icons {
  &__icon {
    @include small-icon;
  }
}
