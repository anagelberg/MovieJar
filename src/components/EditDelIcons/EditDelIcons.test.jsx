import { screen, render } from '@testing-library/react';
import EditDelIcons from './EditDelIcons';
import userEvent from '@testing-library/user-event';

test('renders EditDelIcons component', () => {
    const { container } = render(<EditDelIcons />);
    expect(container.firstChild).toHaveClass('icons');
    expect(container.querySelector('.icons__del')).toBeInTheDocument();
    expect(container.querySelector('.icons__edit')).toBeInTheDocument();
});

test('EditDelIcons component matches snapshot', () => {
    const { container } = render(<EditDelIcons />);
    expect(container.firstChild).toMatchSnapshot();
});

test('calls delClick on delete icon click', () => {
    const delClick = jest.fn();
    const { container } = render(<EditDelIcons delClick={delClick} />);
    userEvent.click(container.querySelector('.icons__del'));
    expect(delClick).toHaveBeenCalled();
});

test('calls editClick on edit icon click', () => {
    const editClick = jest.fn();
    const { container } = render(<EditDelIcons editClick={editClick} />);
    userEvent.click(container.querySelector('.icons__edit'));
    expect(editClick).toHaveBeenCalled();
});