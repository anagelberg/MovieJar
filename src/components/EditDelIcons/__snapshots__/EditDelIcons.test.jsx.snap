// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDelIcons component matches snapshot 1`] = `
<div
  class="icons"
>
  <svg
    class="icons__del"
  >
    delete.svg
  </svg>
  <svg
    class="icons__edit"
  >
    edit.svg
  </svg>
</div>
`;
